kratos/sources/registry.cpp

```diff
@@ -0,0 +1,127 @@
+            auto& item_name = item_path[i];
+            if(p_current_item->HasItem(item_name)){
+                p_current_item = &p_current_item->GetItem(item_name);
+            }
+            else{
+                KRATOS_ERROR << "The item \"" << ItemFullName << "\" is not found in the registry. The item \"" << p_current_item->Name() << "\" does not have \"" << item_name << "\"" << std::endl;
+            }
+        }
+
+        return *p_current_item;
+    }
+
+    void Registry::RemoveItem(std::string const& ItemFullName){
+        const std::lock_guard<LockObject> scope_lock(ParallelUtilities::GetGlobalLock());
+
+        auto item_path = StringUtilities::SplitStringByDelimiter(ItemFullName, '.');
+        KRATOS_ERROR_IF(item_path.empty()) << "The item full name is empty" << std::endl;
+        
+        RegistryItem* p_current_item = &GetRootRegistryItem();
+
+        for(std::size_t i = 0 ; i < item_path.size() - 1 ; i++){
+            auto& item_name = item_path[i];
+            if(p_current_item->HasItem(item_name)){
+                p_current_item = &p_current_item->GetItem(item_name);
+            }
+            else{
+                KRATOS_ERROR << "The item \"" << ItemFullName << "\" is not found in the registry. The item \"" << p_current_item->Name() << "\" does not have \"" << item_name << "\"" << std::endl;
+            }
+        }
+
+        auto& item_name = item_path.back();
+        if(p_current_item->HasItem(item_name)){
+            p_current_item->RemoveItem(item_name);
+        }
+        else{
+            KRATOS_ERROR << "The item \"" << ItemFullName << "\" is not found in the registry. The item \"" << p_current_item->Name() << "\" does not have \"" << item_name << "\"" << std::endl;
+        }
+    }
+
+    bool Registry::HasItem(std::string const& ItemFullName){
+        const std::lock_guard<LockObject> scope_lock(ParallelUtilities::GetGlobalLock());
+
+        auto item_path = StringUtilities::SplitStringByDelimiter(ItemFullName, '.');
+        KRATOS_ERROR_IF(item_path.empty()) << "The item full name is empty" << std::endl;
+        
+        RegistryItem* p_current_item = &GetRootRegistryItem();
+
+        for(std::size_t i = 0 ; i < item_path.size() ; i++){
+            auto& item_name = item_path[i];
+            if(p_current_item->HasItem(item_name)){
+                p_current_item = &p_current_item->GetItem(item_name);
+            }
+            else{
+               return false;
+            }
+        }
+        return true;    
+    }
+
+    std::string Registry::Info() const{
+        return "Registry";
+    }
+
+    void Registry::PrintInfo(std::ostream &rOStream) const{
+        rOStream << Info();
+    }
+
+    void Registry::PrintData(std::ostream &rOStream) const{
+    }
+
+    std::string Registry::ToJson(std::string const& Indentation) const {
+        return GetRootRegistryItem().ToJson(Indentation);
+    }
+
+    RegistryItem& Registry::GetRootRegistryItem(){
```

Comment:
Spaces are mixed here. Not properly aligned

Reply:
Done.